process/transaction/shardProcess.go

```diff
@@ -186,6 +182,11 @@ func (txProc *txProcessor) executingFailedTransaction(
 
 	txProc.txFeeHandler.ProcessTransactionFee(txFee)
 
+	err = txProc.accounts.SaveAccount(acntSnd)
```

Comment:
return directly?

Reply:
no. return is correctly as it is now. as this would be a failed transaction -> InvalidTransaction which error has to be known at transactionpreprocessor.